.github/workflows/ci.yml

```diff
@@ -0,0 +1,134 @@
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Install base libraries
+      uses: input-output-hk/actions/base@latest
+      with:
+        use-sodium-vrf: false
+
+    - name: Install Haskell
+      uses: input-output-hk/actions/haskell@latest
+      id: setup-haskell
+      with:
+        ghc-version: ${{ matrix.ghc }}
+        cabal-version: ${{ matrix.cabal }}
+
+    - name: Update Cabal (Hackage and CHaP)
+      run: |
+        cabal clean
+        cabal update
+
+    # We create a `dependencies.txt` file that can be used to index the cabal
+    # store cache.
+    #
+    # We do not use `plan.json` directly because adding a dependency to our
+    # Cabal files which was already present somewhere else would result in a
+    # diferent plan, even though the set of dependencies is the same.
+    #
+    # In the future we should consider using `cabal-cache` like in the
+    # `cardano-node`'s GitHub workflow.
+    - name: Record dependencies to be used as cache keys
+      id: record-deps
+      run: |
+        cabal build all --dry-run --minimize-conflict-set
+        cat dist-newstyle/cache/plan.json \
+        | jq '.["install-plan"][].id' \
+        | sort \
+        | uniq > dependencies.txt
+
+    # This is used for caching the cabal store.
+    - name: Store month number as environment variable
+      run:  echo "MONTHNUM=$(/usr/bin/date -u '+%m')" >> $GITHUB_ENV
+
+    # Restore the cabal store cache. See `save-cabal-store`.
+    #
+    # When we restore a previous cache and store a new key, we store both files
+    # that were already in the cache, and new files that were created. To
+    # prevent the cache from growing to quickly, we reset the cache each month.
+    #
+    # NOTE: it's vital that all restore-keys include the month number.
+    # Otherwise, we could restore a cache from a previous month, which makes
+    # caches grow unboundedly.
+    - name: "Restore cache for cabal-store"
+      uses: actions/cache/restore@v3
+      id: restore-cabal-store
+      with:
+        path: ${{ steps.setup-haskell.outputs.cabal-store }}
+        key: cabal-store-${{ env.CABAL_CACHE_VERSION }}-${{ runner.os }}-${{ matrix.ghc }}-${{ env.MONTHNUM }}-${{ hashFiles('dependencies.txt') }}
+        restore-keys: |
+          cabal-store-${{ env.CABAL_CACHE_VERSION }}-${{ runner.os }}-${{ matrix.ghc }}-${{ env.MONTHNUM }}-${{ hashFiles('dependencies.txt') }}
+          cabal-store-${{ env.CABAL_CACHE_VERSION }}-${{ runner.os }}-${{ matrix.ghc }}-${{ env.MONTHNUM }}-
+
+    # This is used for caching the `dist-newstyle` directory.
+    - name: Store week number as environment variable
+      run: echo "WEEKNUM=$(/usr/bin/date -u '+%W')" >> $GITHUB_ENV
+
+    # Cache the `dist-newstyle` folder.
+    #
+    # See 'restore-cabal-store' for why we use the week number in the primary
+    # key and restore keys.
+    - name: "Cache dist-newstyle"
+      uses: actions/cache@v3
+      if: matrix.ghc!='8.10.7'
```

Comment:
```suggestion
```

Reply:
The risk here is that we could get the same flaky build errors that we have in `ouroboros-consensus`. But maybe it's fine to try without the condition for now

Reply:
Never mind, I see that we're only using `ghc-9.2.8`!